chainer/optimizer.py

```diff
@@ -27,153 +30,316 @@ def exponential_decay_noise(xp, shape, dtype, hook, opt):
     return xp.random.normal(0, std, shape).astype(dtype)
 
 
-class Optimizer(object):
-    """Base class of all numerical optimizers.
+class Hyperparameter(object):
```

Comment:
Hmm, I used the word "hyperparameter" to represent a single realization of configuration for each update rule/optimizer (i.e. this is the smallest unit of managing hyperparameters in Chainer). Note that we call an array "a parameter" (instead of calling each element of the array a parameter) because array is the smallest unit of managing parameters in Chainer. I don't have a strong opinion in this regard, but it might be better to ask native speakers.